### PR TITLE
docs: correct some statements of code splitting and splitChunks

### PR DESCRIPTION
**Pull request by JSerFeng**
## Summary

Fix some incorrect statements of code splitting and splitChunks. SplitChunks is mainly used for `split` instead of `deduplication`

Remove unnecessary output configuration

Add a description of how name affects the output

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: website/docs/en/guide/optimization/code-splitting.mdx [hunk 1]
```diff
@@ -50,121 +74,69 @@ This will yield the following build result:
...
 };
 
 module.exports = config;
 ```
 
-You should now see that the duplicate modules have been removed from `index.bundle.js` and `another.bundle.js`. Note that the plugin split `shared.js` into `another~index.bundle.js` and removes it from `index.bundle.js` and `another.bundle.js`.
+When rebuild, you will find that `shared.js` has been extracted separately, and there is an additional Chunk in the product that contains `shared.js`.
 
-Build Results:
+### Force the splitting of certain modules into a Chunk.
```
Comment (chenjiahan) [inline website/docs/en/guide/optimization/code-splitting.mdx:116]: Keep title short

```suggestion
### Force the splitting of certain modules
```

File: website/docs/zh/guide/optimization/code-splitting.mdx [hunk 2]
```diff
@@ -50,121 +76,69 @@ console.log('another-module');
...
 };
 
 module.exports = config;
 ```
 
-使用 `optimization.splitChunks` 配置选项之后，现在应该可以看出，`index.bundle.js` 和 `another.bundle.js` 中已经移除了重复的依赖模块。需要注意的是，插件将 `shared.js` 分离到单独的 `another~index.bundle.js` 中 ，并且将其从 `index.bundle.js` 和 `another.bundle.js` 中移除。
+重新打包会发现 `shared.js` 被单独抽离出去，产物中多了一个包含有 `shared.js` 的 Chunk。
 
-构建结果：
+### 强制拆分某些模块到一个 Chunk
```
Comment (chenjiahan) [inline website/docs/zh/guide/optimization/code-splitting.mdx:118]:
```suggestion
### 强制拆分某些模块
```

File: website/docs/en/guide/optimization/code-splitting.mdx [hunk 3]
```diff
@@ -5,17 +5,44 @@ import { ApiMeta } from '@components/ApiMeta';
...
 
-Rspack supports code splitting, which allows splitting the code into other chunks. You have the full control about size of generated assets, which allow you to gain performance improvements in loading time.
+Rspack supports code splitting, which allows splitting the code into other chunks. You have the full control about size and number of generated assets, which allow you to gain performance improvements in loading time.
 
-There are three general approaches to code splitting available:
+## Dynamic import
+
+Rspack use the `import()` syntax that conforms to the ECMAScript proposal for dynamic imports.
+
+:::info Inconsistent behaviors with Webpack
```
Comment (chenjiahan) [inline website/docs/en/guide/optimization/code-splitting.mdx:14]:
```suggestion
:::info Inconsistent behaviors with webpack
```

File: website/docs/zh/guide/optimization/code-splitting.mdx [hunk 4]
```diff
@@ -5,17 +5,46 @@ import { ApiMeta } from '@components/ApiMeta';
 
 # 代码分割
 
-Rspack 支持代码分割特性，允许让你对代码进行分割，控制生成的资源体积来获取资源加载性能的提升。
+Rspack 支持代码分割特性，允许让你对代码进行分割，控制生成的资源体积和资源数量来获取资源加载性能的提升。
 
-常用的代码分离方法有三种：
+这里提出一个概念叫做 Chunk，一个 Chunk 为一个浏览器需要加载的资源。
```
[the remark this answers is not in the document]
Comment (JSerFeng) [reply website/docs/zh/guide/optimization/code-splitting.mdx:10]: Good catch

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**